### PR TITLE
Add scenario focused samples

### PR DESCRIPTION
**Pull request by pvanderknyff**
- Add scenarios folder for samples that are not necessarily full connectors
- Move mysql_odbc and sqlite_extract to scenarios folder
- Remove vendor attributes from postgres_jdbc
- Add new postgres_vendor sample that shows off vendor attributes (same as old postgres_jdbc sample)

Once this is merged, will merge master into dev-2020.3 and fix the delegation sample's folder, as well as add an MCD example

### CODE REVIEW

File: samples/scenarios/vendor_attributes/postgres_vendor_attributes/manifest.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<?xml version='1.0' encoding='utf-8' ?>
```
Comment (rosswbrown) [inline samples/scenarios/vendor_attributes/postgres_vendor_attributes/manifest.xml:1]: Should this postgres_vendor_attributes folder be included?  Looks the same as postgres_vendor.

Comment (pvanderknyff) [reply samples/scenarios/vendor_attributes/postgres_vendor_attributes/manifest.xml]: I renamed the folder, for some reason not all the files were removed. Fixed in latest commit.

File: samples/scenarios/vendor_attributes/postgres_vendor/manifest.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      <customization name="CAP_QUERY_WHERE_FALSE_METADATA" value="yes"/>
+      <customization name="CAP_QUERY_SUBQUERIES_WITH_TOP" value="yes"/>
+      <customization name="CAP_SUPPORTS_SPLIT_FROM_LEFT" value="yes"/>
+      <customization name="CAP_SUPPORTS_SPLIT_FROM_RIGHT" value="yes"/>
+      <customization name="CAP_SUPPORTS_UNION" value="yes"/>
+      <customization name="CAP_QUERY_ALLOW_PARTIAL_AGGREGATION" value="no"/>
+      <customization name="CAP_QUERY_TIME_REQUIRES_CAST" value="yes"/>
+    </customizations>
+  </connection-customization>
+  <connection-dialog file='connection-dialog.tcd'/>
```
Comment (rosswbrown) [inline samples/scenarios/vendor_attributes/postgres_vendor/manifest.xml:33]: Some day let's rename all the connection-dialog.tcd to connectionDialog.tcd :)

File: samples/scenarios/vendor_attributes/postgres_vendor/dialect.tdd [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<dialect name='SimplePostgres'
+         class='postgres_vendor'
+         base='PostgreSQL90Dialect'
+         version='18.1'>
+   <function-map>
```
Comment (rosswbrown) [inline samples/scenarios/vendor_attributes/postgres_vendor/dialect.tdd:6]: For the scenario based ones we can just use the base dialect as is.  This FLOAT was provided as an example originally.

**Pull request merged**